Aj/top nav

ajsingh273 opened the pull request:
@calumhalcrow I made updates to the topbar from [Stuart’s feedback](https://www.pivotaltracker.com/story/show/91363330)

Most of the changes were pretty straightforward. The last item is a little tricky because of a “fixed" topbar. He wants the mobile menu to function similar to this [example](http://rosenelectronics.com/seatback-systems.htm), by pushing the page down. But, the example uses a static topbar that scrolls with the page. I think its best to leave it as is, with the fold. But please let me know if you have any suggestions on this. I found some [script](http://jsfiddle.net/degoeym/2u3u6ejh/) that may accomplish this, but was not able to make it work.

I also added a small note to each of the docs, indicating how often endpoints get updated. This was a earlier story, but the note got lost with the new v2 docs. Erik suggested that we put this back in. I think the frequency is correct for all of them, but please confirm.

Thanks.

ajsingh273 committed:
top bar and menu updates

ajsingh273 committed:
added update frequency and some styles

ajsingh273 committed:
updated page responsive size

ajsingh273 committed:
updated frequency note in docs

ajsingh273 committed:
some padding to mobile menu

ajsingh273 committed:
fixed style issue for firefox

calumhalcrow commented on _includes/footer.html:
```diff
@@ -8,10 +8,13 @@
 
+<script src="http://code.jquery.com/jquery-2.1.0.js"></script>
 <script src="{{ site.baseurl }}/javascripts/scale.fix.js"></script>
 <script src="http://code.jquery.com/jquery-1.8.3.min.js"></script>
```
You're loading two versions of jQuery. You don't want to do that.

calumhalcrow commented on _posts/v2/2014-11-12-ita-faqs.markdown:
```diff
@@ -7,6 +7,8 @@ published: true
 
 #FAQs on Exporting API
 
+######Data set updated monthly
```
We don't import these periodically at present, so it's technically not correct to say this.

calumhalcrow commented on _posts/v2/2015-02-23-tariff-rates.markdown:
```diff
@@ -7,6 +7,8 @@ published: true
 
 #Tariff Rates API
 
+######Data set updated yearly
```
Again, we don't actually import these on any schedule.

calumhalcrow commented on _posts/v2/2015-02-24-ita-office-locations.md:
```diff
@@ -7,6 +7,8 @@ published: true
 
 #ITA Offices & Centers API
 
+######Dataset updated monthly
```
We don't import these periodically.

calumhalcrow commented on _posts/v2/2015-02-24-market-research-library.md:
```diff
@@ -5,7 +5,9 @@ title: Market Research Library API
-#Market Research Library API
+#Market Research Library API 
+
+######Data set updated weekly
```
We don't import these periodically.

calumhalcrow commented:
We import the following consolidated sources daily. Everything else is not import periodically.

CSL
Trade Events
Trade Leads
Trade Articles

You can see the Chef recipe which installs the crontab [here](https://github.com/GovWizely/webservices-cookbooks/blob/master/gw_webservices/recipes/crontabs.rb).

calumhalcrow commented on stylesheets/bootstrap.min.css:
```diff
@@ -1257,7 +1257,7 @@ hr {
 }
 
-@media (min-width:768px) {
+@media (min-width:1300px) {
```
The fact that you've edited the bootstrap source file makes me raise my eyebrow.

calumhalcrow replied:
I'm not sure if this change is directly causing this, but have some issues with how the nav behaves when the browser width is changed:
1. The "burger" icon appears way too soon (i.e. when the page is still much wider than it would be on a handheld device). It's purpose is to help touchscreen users, but it will appear for many desktop/laptop users also. Look at this screenshot of my entire 15" screen:

![screen shot 2015-05-04 at 3 28 34 pm](https://cloud.githubusercontent.com/assets/277504/7450006/615be620-f272-11e4-88d8-78ebc4e37ada.png)
1. The "burger" icon sometimes disappears to the right. Again, I think it's to do with the 1300px change but haven't verified.

![screen shot 2015-05-04 at 3 31 07 pm](https://cloud.githubusercontent.com/assets/277504/7450076/3438cba8-f273-11e4-9da4-d85c335a9964.png)

As I mentioned in another comment, I think you shouldn't mess with the Bootstrap implementation and just use it as is. If it ain't broke...

calumhalcrow commented on _includes/footer.html:
```diff
@@ -8,10 +8,13 @@
+<script src="http://code.jquery.com/jquery-2.1.0.js"></script>
 <script src="{{ site.baseurl }}/javascripts/scale.fix.js"></script>
 <script src="http://code.jquery.com/jquery-1.8.3.min.js"></script>
 <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
```
In fact, there are three versions of jQuery being loaded! I'll fix this.

calumhalcrow commented on _includes/footer.html:
```diff
@@ -25,4 +28,4 @@
 <script src="{{ site.baseurl }}/javascripts/jquery-xdomainrequest.js"></script>
```
Ok, that's four (see line above).

calumhalcrow commented:
@ajsingh273 regarding the behavior of the top nav: doesn't the [Bootstrap website](http://getbootstrap.com/) also exhibit the behavior you want? If you follow the markup conventions there (nav HTML, class names, etc.) you'll be able to get the same thing here (given that you're using Bootstrap already).

BTW, if you want to keep the top nav fixed on the page, you can't have the dropdown push the content down. It's impossible if you think about it: the nav is fixed, so it can never be scrolled past to get to the content below. I don't like the fixed nav BTW, especially when the width is narrow. On a mobile device, you always want to maximize screen real-estate, and always having nav on the page goes against that principle. My advice is: implement the nav exactly as Bootstrap does it. It's never going to be better than how they've implemented it. (Their team are experts and have thought long and hard about the best way to do this kind of thing. Do you think you guys can do better? I know I couldn't :P ).

ajsingh273 commented:
@calumhalcrow Yes, my intention is to just use the Bootstrap library out of the box.  

I agree, logically its not possible to have a fixed top navigation and have it push the page down. The reason for the “fixed" nav is because it fits with the rest of the page layout (that is, if we continue using this layout). The “default" and “static" nav would not work here.

ajsingh273 committed:
removed extra jQuery and API frequency notes

ajsingh273 commented:
@calumhalcrow 
- Removed the extra jQuery files 
- Updated the frequency notes in each of the docs. I did look at the crontabs file, but was not sure about the others: MRL, Office Locations, FAQs, Tariff Rates. 
- The mobile width change to 1300px was because Stuart requested to have the page show the mobile hamburger immediately after the first screen shift.

calumhalcrow commented:
I don't think it makes sense to have mobile-centric nav kick in at 1300px width, but I'm not going to argue about it either.

ajsingh273 committed:
removing 1300px width

calumhalcrow commented on stylesheets/styles.css:
```diff
@@ -547,6 +547,23 @@ footer {
+padding-left:25px;
+}
+=======
+>>>>>>> parent of c3c8bd0... updated page responsive size
```
You have accidentally pushed a merge conflict here.

ajsingh273 committed:
improved mobile width

ajsingh273 commented:
@calumhalcrow  Firstly, I undid the 1300px width which was related to one of Stuart’s earlier requests. We chatted yesterday, and he was ok with going back to the original media queries. So now, everything should be back to Bootstrap standards. 

The second part was matching the body width to the header, since we are not using a grid system. So instead, I applied the container class around the wrapper, which seemed to work well. 

Again, the reason for a [fixed](http://getbootstrap.com/examples/navbar-fixed-top/) top nav versus a [default](http://getbootstrap.com/examples/navbar/), is so we can continue using the side nav layout. Also, it leaves some branding on the top of the page. 

Please let me know if these updates work for you. 

Thanks.
